packages/jest-runner/fuzz.ts

```diff
@@ -64,15 +68,14 @@ export const fuzz: FuzzTest = (name, fn) => {
 	// points to the element containing the fuzz function.
 	const testStatePath = currentTestStatePath(testName);
 
-	// The timeout setting is extracted from the test state and defaults to 5s.
-	// Setting timeouts on this level is necessary, as they apply to the
-	// individual inputs and not the whole run.
-	const timeout = currentTimeout();
-
 	const corpus = new Corpus(testFile, testStatePath);
 
 	const fuzzingConfig = loadConfig();
-	fuzzingConfig.timeout = timeout;
+
+	// Timeout priority is: test timeout > config timeout > default timeout.
+	if (!timeout)
+		if (fuzzingConfig.timeout) timeout = fuzzingConfig.timeout;
```

Comment:
We pass the timeout option in the fuzzing config as the `-timeout` flag for libFuzzer (See `packages/core/core.ts:295`). To make this consistent, we should instead configure the `-max_total_time` to indicate the maximal total time in seconds to run the fuzzer. Could you update this there?

Reply:
True, we should separate the two settings. `timeout` one invocation, `max_total_time` whole fuzzing run (not relevant in regression mode).

Reply:
I disagree, I think "timeout" should mean the same thing (timeout for one execution of the fuzz function) in regression and in fuzzing mode. If it doesn't, you have to update all your *.fuzz.js files every time you switch the mode.
E.g.:
```
it.fuzz("confused user", (data) => {
   ...
}, 1000));
```
"1000" in regression mode is what a Jest-user would expect---each regression test takes 1 second.
Should they switch into fuzzing mode, suddenly, the same "1000" means  that the whole fuzzing run will take 1 second, and the timeout per input is set to libFuzzer's default 1200 seconds.
Please check the current version :wink:

Reply:
I think we all mean the same. `timeout` should always configure the timeout for one invocation. The current version looks like it's doing exactly that.